00-framework-01-maturity.md

```diff
@@ -0,0 +1,118 @@
+their evolution and adoption within the community, while maximising consensus
+around a common lifecycle and criteria for progression across stages.
+
+We propose a five-stage process to classify the maturity of a specification in
+terms of completeness, demonstrability of implementation, community
+acceptance, and level of technical detail. It is somewhat inspired in the W3C
+Process [0].
+
+  1. `Working Draft`
+  2. `Candidate Recommendation`
+  3. `Recommendation`
+  4. `Deprecated`
+  5. `Abandoned Draft`
+
+## Stage 1: Working draft
+
+The specification of the system, process, protocol or item is under active
+development. This stage is lightweight and mostly self-directed by the author.
+We aim to reduce the barrier to entry, and it's designed to allow for
+iterative experimentation, discovery and pivoting.
+
+We don't enforce a hard template in an attempt to enhance author
+expressability and creativity.
+
+We enter this stage by posting an `Initial Working Draft` that covers:
+  * context: what is the current situation or a brief overview of the
+    environment the specification targets.
+  * motivation: why this specification is relevant, and how it advances the
+    status quo.
+  * scope and rationale: what areas of the technical system the specification
+    impacts.
+  * goals: what we expect to achieve (positively and negatively) as a result
+    of implementing the specification.
+  * expected feature set: a summary/enumeration of features the spec provides.
+  * tentative technical directions: how are we planning to materialise the
+    specification in terms of system design.
+
+The `Initial Working Draft` shall be reviewed by the community in no more than 3
+working days. Should there be no defects in form, content or serious technical
+soundness issues, the `Initial Working Draft` will be accepted and merged.
+
+Ideas deemed controversial or breaking, and those that garner subjective
+opposition, will still be accepted in order to give them a venue to grow,
+mature and iterate.
+
+Once the `Initial Working Draft` is merged, the author may continue revising
+and evolving their specification by self-approving their own *Pull Requests*.
+
+We MAY use GitHub's
+[`CODEOWNERS`](https://help.github.com/en/articles/about-code-owners) feature
+to model the approval policy.
+
+## Stage 2: Candidate Recommendation
+
+The changes requested by the specification are considered plausible and
+desirable.
+
+The specification document itself is technically complete. It defines wire
+level formats for interoperability, error codes, algorithms, data structures,
+heuristics, behaviours, etc., in a way that it is sufficient to enable
+contributors to develop an interoperable implementation.
+
+There is at least ONE implementation conforming to the specification. That
+implementation serves as the Reference Implementation.
+
+The promotion from a `Working Draft` to a `Candidate Recommendation` is done
```

Comment:
What is the criterium for achieving this? Does everyone need to agree?

Reply:
Gonna answer more generically in the pull requests, as other questions too relate to consensus and the review process.

Reply:
https://github.com/libp2p/specs/pull/169#issuecomment-494459635